src/components/common/DiscountCode.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+  <div class="row">
+    <div class="col-sm-offset-9 col-sm-3">
+      <ServerError :error="serverError">
+        <template slot-scope="{ graphQLError }">
+          {{ getErrorMessage(graphQLError) }}
+        </template>
+      </ServerError>
+    </div>
+    <div class="col-sm-12">
+      <form @submit.prevent="submitDiscountCode"
+            class="text-right promotion-info">
+        <span class="text-uppercase promo-info-text">
+          {{ $t('code') }}
+          <img src="../../assets/img/information.png"
+               class="info-icon"
+               alt="information icon">
+        </span>
+        <input id="promo-code" type="text" name="code" v-model="code"
+               data-test="discount-code-input">
```

Comment:
As we use vuelidate (https://vuelidate.netlify.com/) in Sunrise, we implement forms a bit different.

Check a form and see how inputs are written (without name and id -if not being used for style), specifying validators (required field), showing validation errors, not allowing to submit invalid forms, etc.